### ISSUE TITLE
SA1202 doesn't work in record types

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#3693]
A C# `record` with declared properties doesn't fire SA1202 when an `internal` property appears before a `public` property.

Given this type declaration:

```cs
record A
{
	internal bool B { get; }

	public bool C { get; }
}
```

I expect an SA1202 warning, but none appears.
Change the `record` to `class`, and the warning appears.

StyleCop.Analyzers 1.2.0.507

**Comment by bjornhellander** [comment]
It looks like this rule hasn't been updated at all due to `record` and `record struct`. It seems it simply ignores them, both themselves and their members. So this also does not trigger SA1202:
```
internal record A {
}
public record B { 
}
```

**Comment by bjornhellander** [comment]
Is the expected behaviour to treat records and classes as if they were the same? And similarly for record structs and structs?

**Comment by sharwell** [comment]
@bjornhellander I submitted a pull request to add support for members of interfaces (accessibility modifiers supported starting in C# 8) and records (`record` in C# 9, `record class` and `record struct` in C# 10).
